lib/components/account/account_body.dart

```diff
@@ -0,0 +1,178 @@
+        ),
+      ),
+      body: Container(
+        child: Column(
+          children: <Widget> [
+            PageHeader("Account"),
+            StreamBuilder(
+              stream: _authService.signedInUser,
+              builder: (context, snapshot) {
+                if (!snapshot.hasData) {
+                  return LoadingIndicator();
+                }
+                final User _userData = snapshot.data;
+                return Row(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  children: <Widget>[
+                    Expanded(
+                      flex: 1,
+                      child: Icon(
+                        Icons.mail,
+                      ),
+                    ),
+                    Expanded(
+                      flex: 4,
+                      child: TextFormField(
+                        initialValue: _userData.email,
+                        enabled: false,
+                        decoration: InputDecoration(
+                          labelText: "Email",
+                        ),
+                      ),
+                    ),
+                    Expanded(
+                      flex: 1,
+                      child: SizedBox(height: 10,)
+                    ),
+                  ],
+                );
+              }
+            ),
+            StreamBuilder(
+            stream: _userService.userData,
+            builder: (context, snapshot) {
+                if (!snapshot.hasData) {
+                  return LoadingIndicator();
+                }
+                final UserData _userData = snapshot.data;
+                return Column(
+                    children: <Widget> [
+                      Row(
+                        mainAxisAlignment: MainAxisAlignment.center,
+                        children: <Widget>[
+                          Expanded(
+                            flex: 1,
+                            child: Icon(
+                              Icons.face,
+                            ),
+                          ),
+                          Expanded(
+                            flex: 4,
+                            child: TextFormField(
+                              onChanged: (value) {
+                                name = value;
+                              },
+                              initialValue: _userData.displayName,
+                              enabled: _nameActiveStatus,
+                              decoration: InputDecoration(
+                                labelText: "Name",
+                              ),
+                            ),
+                          ),
+                          Expanded(
+                            flex: 1,
+                            child: IconButton(
+                              icon: (_nameActiveStatus ? Icon(Icons.check) : Icon(Icons.edit)),
+                              onPressed: () {
+                                setState(() {
```

Comment:
Clicking the edit buttons for the form fields rebuilds the entire page.  Need to make it so that only the target form field is affected by it.  Add a print statement within the build method for the page and click the buttons to see it happen.

Reply:
Or just use a different method